docs/src/proposals/rip-curl.md

```diff
@@ -0,0 +1,56 @@
+# RiP Curl: low-latency, transaction-oriented RPC
+
+## Problem
+
+Solana's initial RPC implementation was created for the purpose of allowing
+users to confirm transactions that had just recently been sent to the cluster.
+It was designed with memory usage in mind such that any validator should be
+able to support the API without concern of DoS attacks.
+
+Later down the line, it became desirable to use that same API to support the
+Solana explorer. The original design only supported minutes of history, so we
+changed it to instead store transaction statuses in a local RocksDB instance
+and offer days of history. We then extended that to 6 months via BigTable.
+
+With each modification, the API became more suitable for explorers and less
+appealing for transaction processing. The clients poll for transaction status
+instead of being notified, giving the false impression of higher confirmation
+times. Furthermore, what clients can poll for is limited, preventing them from
+making reasonable real-time decisions, such as recognizing a transaction is
+confirmed as soon as particular, trusted validators vote on it.
+
+## Proposed Solution
+
+A web-friendly, transaction-oriented, streaming API built around the
+validator's ReplayStage.
+
+Improved client experience:
+
+* Support connections directly from JavaScript apps.
+* Clients can be notified of confirmation in optimal time.
+* Clients can be notified in real-time of details such as votes and the voter's
```

Comment:
We don't have a way to subscribe to real-time votes, but we do allow subscribing to account changes for all (or specific) vote accounts

Reply:
True, that's a good workaround. When I think of a streaming API, I think of how *commitment* increases as votes roll in, ("commitment" as I had originally defined it, `stake * lockout`). As a user, I want to know how much money validators will lose if they jump to a fork that doesn't include my transaction. That's different than "confirmations", but I think commitment adds precision and perhaps if we put more emphasis on that, users would gravitate toward it.

Reply:
We do actually have vote subscriptions: https://docs.solana.com/apps/jsonrpc-api#votesubscribe
However, they do not currently return any information about the voter. It looks like voter pubkey and stake information are both available in the relevant method in cluster_info_vote_listener, so it would be trivial to add.